📖 Add In-place upgrades feature group (#9559)
kubernetes-sigs/cluster-api: docs/community/20231016-in-place-upgrades.md

## Scope

The scope of this effort will be the following:

1. Define the scope of what In-place upgrades means in the context of Cluster API.
2. Write a CAPI proposal with the recommended solution.

## Communication

We will meet on [TODO: agree on a day and time][zoomMeeting]. [Convert to your timezone](http://www.thetimezoneconverter.com/?t=09:00&tz=PT%20%28Pacific%20Time%29). Meeting notes will be documented in this [TODO: create Google Doc]().

Regular, summarized updates of group progress will be provided during weekly Cluster API office hours on Wednesdays @ 10:00 PT.

Chat with stakeholders on Kubernetes [Slack](http://slack.k8s.io/) in the [cluster-api](https://kubernetes.slack.com/archives/C8TSNPY4T) channel.

## Stakeholders

Primary Stakeholders are listed below:

- Dharmjit Singh (@dharmjit, VMware)

> - Guillermo Gaston (@g-gaston, AWS)

Reviewer: Could you add @sdodson and myself to this list?

---

> title: Feature Group for Managed Kubernetes in Cluster API

Reviewer 2: ```suggestion
title: Feature Group for In-place Upgrades support in Cluster API
```

authors:
  - "@g-gaston"
reviewers:
    - "@dharmjit"
    - "@vincepri"

>     - "@sbueringer”"

Reviewer 2: typo
```suggestion
    - "@sbueringer"
```

## Scope

The scope of this effort will be the following:

1. Define the scope of what In-place upgrades means in the context of Cluster API.
2. Write a CAPI proposal with the recommended solution.

## Communication

We will meet on [TODO: agree on a day and time][zoomMeeting]. [Convert to your timezone](http://www.thetimezoneconverter.com/?t=09:00&tz=PT%20%28Pacific%20Time%29). Meeting notes will be documented in this [TODO: create Google Doc]().

Regular, summarized updates of group progress will be provided during weekly Cluster API office hours on Wednesdays @ 10:00 PT.

Chat with stakeholders on Kubernetes [Slack](http://slack.k8s.io/) in the [cluster-api](https://kubernetes.slack.com/archives/C8TSNPY4T) channel.

## Stakeholders

Primary Stakeholders are listed below:

> - Alexander Demicev (@@alexander-demicev, SUSE)

Author: 🤦🏻

## User Story and Problem Statement

As a CAPI Kubernetes cluster admin I want to upgrade the k8s components (for example, a minor Kubernetes upgrade) of my cluster without replacing the machines.

At present, CAPI has "immutable infrastructure" as one of its axioms: once created, Machines are not updated, they are just replaced (a new one is created and old one is deleted). As a consequence, the only supported upgrade strategy is rolling update. However, for certain use cases (such as Single-Node Clusters with no spare capacity, Multi-Node Clusters with VM/OS customizations for high-performance/low-latency workloads or dependency on local persistent storage), upgrading a cluster via RollingUpdate strategy could either be not feasible or a costly operation (requiring to re-apply customizations on newer nodes and hence more downtime).

## Scope

The scope of this effort will be the following:

1. Define the scope of what In-place upgrades means in the context of Cluster API.
2. Write a CAPI proposal with the recommended solution.

## Communication

> We will meet on [Wednesdays at 08:00 PT (Pacific Time)][zoomMeeting]. [Convert to your timezone](https://dateful.com/time-zone-converter?t=09:00&tz=PT%20%28Pacific%20Time%29). Meeting notes will be documented in this [Google Doc](https://docs.google.com/document/d/1GmRd6MyQ0mWAoJV6rCHhZTSTtKMKHdJzhXm0BLBXOnw).

Reviewer 3: I think you are missing to add `zoomMeeting` reference at the end.

> Regular, summarized updates of group progress will be provided during weekly Cluster API office hours on Wednesdays @ 10:00 PT.

Reviewer 3: ```suggestion
Regular, summarized updates of group progress will be provided during weekly Cluster API office hours on Wednesdays @ 10:00 PT on [Zoom][zoomMeeting].
```

Chat with stakeholders on Kubernetes [Slack](http://slack.k8s.io/) in the [cluster-api](https://kubernetes.slack.com/archives/C8TSNPY4T) channel.

## Stakeholders

Primary Stakeholders are listed below:

- Mayur Das (@mayur-tolexo, VMware)
- Alexander Demicev (@alexander-demicev, SUSE)
- Scott Dodson (@sdodson, Red Hat)
- Guillermo Gaston (@g-gaston, AWS)
- Furkat Gofurov (@furkatgofurov7, SUSE)

> - Dharmjit Singh (@dharmjit, VMware)

Reviewer 3: ```suggestion
- Dharmjit Singh (@dharmjit, VMware)

[zoomMeeting]: https://zoom.us/j/861487554
```
